integration/querier_remote_read_test.go

```diff
@@ -162,118 +165,230 @@ func isEqualSampleAndHistogram(t *testing.T, expectedVector *model.Sample, histo
+			stream := remote.NewChunkedReader(httpResp.Body, remote.DefaultChunkedReadLimit, nil)
+
+			var results []prompb.ChunkedReadResponse
+			for {
+				var res prompb.ChunkedReadResponse
+				err := stream.NextProto(&res)
+				if errors.Is(err, io.EOF) {
+					break
+				}
+				require.NoError(t, err)
+				results = append(results, res)
+			}
+
+			// Validate the returned remote read data
+			i := 0
+			for _, result := range results {
+				// We're only expected a single series `series_1`.
+				require.Len(t, result.ChunkedSeries, 1)
+				require.Equal(t, "series_1", result.ChunkedSeries[0].Labels[0].GetValue())
+
+				for _, rawChk := range result.ChunkedSeries[0].Chunks {
+					var enc chunkenc.Encoding
+					switch rawChk.Type {
+					case prompb.Chunk_XOR:
+						enc = chunkenc.EncXOR
+					case prompb.Chunk_HISTOGRAM:
+						enc = chunkenc.EncHistogram
+					// TODO: Once prompb.Chunk_FLOAT_HISTOGRAM gets added in https://github.com/prometheus/prometheus/pull/12085 we can switch to using that
+					case prompb.Chunk_Encoding(3):
+						enc = chunkenc.EncFloatHistogram
+					default:
+						require.Fail(t, "unrecognized chunk type")
+					}
+
+					chk, err := chunkenc.FromData(enc, rawChk.Data)
+					require.NoError(t, err)
+
+					chkItr := chk.Iterator(nil)
+					j := 0
+					for valType := chkItr.Next(); valType != chunkenc.ValNone; valType = chkItr.Next() {
+						switch valType {
+						case chunkenc.ValFloat:
+							ts, val := chkItr.At()
+							if samples != nil {
+								require.Equal(t, samples[i].Timestamp, ts)
+								require.Equal(t, samples[i].Value, val)
+							}
+						case chunkenc.ValHistogram:
+							ts, h := chkItr.AtHistogram()
+							if histograms != nil {
+								require.Equal(t, histograms[i].Timestamp, ts)
+
+								expected := remote.HistogramProtoToHistogram(histograms[i])
+								if j > 0 {
+									// This is set when the histogram is returned for any sample that isn't the first sample in the chunk.
+									// https://github.com/prometheus/prometheus/blob/001ee2620e094970e5657ce39275b2fccdbd1359/tsdb/chunkenc/histogram_meta.go#L463-L466
+									expected.CounterResetHint = histogram.NotCounterReset
+								}
+								require.Equal(t, expected, h)
+							}
+						case chunkenc.ValFloatHistogram:
+							ts, fh := chkItr.AtFloatHistogram()
+							if histograms != nil {
+								require.Equal(t, histograms[i].Timestamp, ts)
+
+								expected := remote.HistogramProtoToFloatHistogram(histograms[i])
+								if j > 0 {
+									// This is set when the histogram is returned for any sample that isn't the first sample in the chunk.
+									// https://github.com/prometheus/prometheus/blob/001ee2620e094970e5657ce39275b2fccdbd1359/tsdb/chunkenc/histogram_meta.go#L463-L466
+									expected.CounterResetHint = histogram.NotCounterReset
+								}
+								require.Equal(t, expected, fh)
+							}
+						default:
+							require.Fail(t, "unrecognized value type")
+						}
+						i++
+						j++
```

Comment:
Are there more descriptive names for `i` and `j` we could use? Their purpose wasn't immediately clear to me.

Reply:
Good idea; changed `i` to `sampleIdx` and `j` to `chkIdx`.